alleycats-core/src/main/scala/alleycats/std/iterable.scala

```diff
@@ -65,6 +65,18 @@ trait IterableInstances {
       override def traverse[G[_], A, B](fa: Iterable[A])(f: A => G[B])(implicit G: Applicative[G]): G[Iterable[B]] =
         if (fa.isEmpty) G.pure(Iterable.empty)
         else G.map(Chain.traverseViaChain(toImIndexedSeq(fa))(f))(_.toVector)
+
+      override def mapAccumulate[S, A, B](init: S, fa: Iterable[A])(f: (S, A) => (S, B)): (S, Iterable[B]) = {
```

Comment:
What about also adding `mapWithIndex` and `zipWithIndex` here?

```scala
override def mapWithIndex[A, B](fa: Iterable[A])(f: (A, Int) => B): Iterable[B] =
  fa.zipWithIndex.map(ai => f(ai._1, ai._2))

override def zipWithIndex[A](fa: Iterable[A]): Iterable[(A, Int)] =
  fa.zipWithIndex
```

The motivation being they would preserve their underling class.

Reply:
I made this change but I'm suspect about the trying to add invariants we don't really promise (returning the same underlying type).

1. this is already in alleycats where there are reasons to believe things are suspect. In this case Iterable is so weak it doesn't necessarily iterate in any meaningful order (otherwise it would be Seq).
2. when people start to assume we offer some stronger contracts, they often wind up getting into trouble and file issues and then we are in a jam. For instance, mapWithIndex is exercising a different code path from mapAccumulate here, and if they want some other law we don't promise (like maybe they do a cast at the end because they assume the Iterable is of the same type), then we get into weird issues of explaining why we went out of our way to maintain the class, but sometimes fail to.


anyway, it's alleycats, so I added it. But I hope people won't rely on it.